flink-runtime/src/test/java/org/apache/flink/runtime/operators/util/OutputEmitterTest.java

```diff
@@ -48,404 +49,272 @@
-		@SuppressWarnings({"unchecked", "rawtypes"})
-		final TypeComparator<Record> doubleComp = new RecordComparatorFactory(new int[] {0}, new Class[] {DoubleValue.class}).createComparator();
-		final ChannelSelector<SerializationDelegate<Record>> oe1 = new OutputEmitter<Record>(ShipStrategyType.PARTITION_HASH, doubleComp);
-		final SerializationDelegate<Record> delegate = new SerializationDelegate<Record>(new RecordSerializerFactory().getSerializer());
-		
-		
-		;
-		
-		Record rec = null;
-		
-		try {
-			PipedInputStream pipedInput = new PipedInputStream(1024*1024);
-			
-			DataInputView in = new DataInputViewStreamWrapper(pipedInput);
-			DataOutputView out = new DataOutputViewStreamWrapper(new PipedOutputStream(pipedInput));
-			
-			rec = new Record(1);
-			rec.setField(0, new IntValue());
-			
-			rec.write(out);
-			rec = new Record();
-			rec.read(in);
-		} catch (IOException e) {
-			fail("Test erroneous");
-		}
+		final TypeComparator<Record> doubleComp = new RecordComparatorFactory(
+			new int[] {0}, new Class[] {DoubleValue.class}).createComparator();
+		final ChannelSelector<SerializationDelegate<Record>> selector = new OutputEmitter<>(
+			ShipStrategyType.PARTITION_HASH, doubleComp);
+		final SerializationDelegate<Record> delegate = new SerializationDelegate<>(new RecordSerializerFactory().getSerializer());
+
+		PipedInputStream pipedInput = new PipedInputStream(1024 * 1024);
+		DataInputView in = new DataInputViewStreamWrapper(pipedInput);
+		DataOutputView out = new DataOutputViewStreamWrapper(new PipedOutputStream(pipedInput));
+
+		Record record = new Record(1);
+		record.setField(0, new IntValue());
+		record.write(out);
+		record = new Record();
+		record.read(in);
 
 		try {
-			delegate.setInstance(rec);
-			oe1.selectChannels(delegate, 100);
+			delegate.setInstance(record);
+			selector.selectChannels(delegate, 100);
 		} catch (DeserializationException re) {
 			return;
 		}
 		Assert.fail("Expected a NullKeyFieldException.");
 	}
-	
-	@SuppressWarnings({"serial", "rawtypes"})
+
+	private boolean verifyWrongPartitionHashKey(int position, int fieldNum) {
+		final TypeComparator<Record> comparator = new RecordComparatorFactory(
+			new int[] {position}, new Class[] {IntValue.class}).createComparator();
+		final ChannelSelector<SerializationDelegate<Record>> selector = new OutputEmitter<>(
+			ShipStrategyType.PARTITION_HASH, comparator);
+		final SerializationDelegate<Record> delegate = new SerializationDelegate<>(new RecordSerializerFactory().getSerializer());
+
+		Record record = new Record(2);
+		record.setField(fieldNum, new IntValue(1));
+		delegate.setInstance(record);
+
+		try {
+			selector.selectChannels(delegate, 100);
+		} catch (NullKeyFieldException re) {
+			Assert.assertEquals(position, re.getFieldNumber());
+			return true;
+		}
+		return false;
+	}
+
+	private int[] getSelectedChannels(
+		ShipStrategyType shipStrategyType,
+		int numRecords,
+		int numChannels,
+		Either<Integer, String> recordType) {
```

Comment:
formatting nit: could you add one more tab before the parameters? As it is now, parameters are fusing with the method code.